src/main/java/chungus/Chungus.java

```
@@ -244,10 +244,219 @@ private void addEvent(String input) throws ChungusException {
         ui.showTaskAdded(t, tasks.size());
     }
 
+    /**
+     * Processes a user input and returns a response string for GUI display.
+     * This method replaces the console-based run() method for GUI usage.
+     *
+     * @param input User input command
+     * @return Formatted response string
+     */
+    public String getResponse(String input) {
+        if (input.trim().equals(CommandType.BYE.getCommand())) {
+            saveTasksToStorage();
+            return ui.getByeMessage();
+        }
+        try {
+            CommandType command = parser.parseCommandType(input);
+            if (command == null) {
+                if (input.trim().isEmpty()) {
+                    return ""; // ignore silently
+                } else {
+                    throw new ChungusException("I'm sorry, but I don't know what that means :-(");
+                }
+            } else {
+                return processCommandForGui(command, input);
+            }
+        } catch (ChungusException e) {
+            return ui.getErrorMessage(e.getMessage());
+        } catch (Exception e) {
```

Hi! I would not recommend broad catch of Exception as it hides the real error during debugging and testing, would recommend removing the broad catch of Exception and let the program crash if there really is one based on instructions of CS2103. If we insist on catching it, I would suggest throwing a RuntimeException and throw e instead.

---

```suggestion
```